main.nf

```diff
@@ -0,0 +1,932 @@
+                   saveAs: { params.saveReference ? it : null }, mode: 'copy'
+
+        input:
+        file fasta from fasta_for_bwa_index
+
+        output:
+        file "*.{amb,ann,bwt,pac,sa}" into bwa_index
+
+        script:
+        """
+        bwa index $fasta
+        """
+    }
+    //Create a FastA index for non-indexed genomes
+    process makeFastaIndex {
+        tag "$params.gfasta"
+        publishDir path: { params.saveReference ? "${params.outdir}/reference_genome" : params.outdir },
+                   saveAs: { params.saveReference ? it : null }, mode: 'copy'
+        
+        input:
+        file fasta from fasta_for_samtools_index
+
+        output:
+        file "*.fai" into samtools_index
+
+        script:
+        """
+        samtools faidx $fasta
+        """
+    }
+} else {
+    bwa_index = Channel.fromPath("${params.bwa_index}")
+}
+
+
+/*
+* STEP 0 - FastQC
+*
+*/
+
+process fastqc {
+    tag "$name"
+    publishDir "${params.outdir}/fastqc", mode: 'copy',
+        saveAs: {filename -> filename.indexOf(".zip") > 0 ? "zips/$filename" : "$filename"}
+
+    input:
+    set val(name), file(reads) from read_files_fastqc
+
+    output:
+    file '*_fastqc.{zip,html}' into fastqc_results
+    file '.command.out' into fastqc_stdout
+
+    script:
+    """
+    fastqc -q $reads
+    """
+}
+
+
+/*
+ * STEP 1 - trim with trim galore
+ */
+
+if(params.notrim){
+    trimmed_reads = read_files_trimming
+    trimgalore_results = []
+} else {
+    process trim_galore {
+        tag "$name"
+        publishDir "${params.outdir}/trim_galore", mode: 'copy'
+
+        input:
+        set val(name), file(reads) from read_files_trimming
+
+        output:
+        set val(name), file('*fq.gz') into trimmed_reads
+        file '*trimming_report.txt' into trimgalore_results, trimgalore_logs
+
+        script:
+        single = reads instanceof Path
+        c_r1 = params.clip_r1 > 0 ? "--clip_r1 ${params.clip_r1}" : ''
+        c_r2 = params.clip_r2 > 0 ? "--clip_r2 ${params.clip_r2}" : ''
+        tpc_r1 = params.three_prime_clip_r1 > 0 ? "--three_prime_clip_r1 ${params.three_prime_clip_r1}" : ''
+        tpc_r2 = params.three_prime_clip_r2 > 0 ? "--three_prime_clip_r2 ${params.three_prime_clip_r2}" : ''
+        if (params.singleEnd) {
+            """
+            trim_galore --gzip $c_r1 $tpc_r1 $reads
+            """
+        } else {
+            """
+            trim_galore --paired --gzip $c_r1 $c_r2 $tpc_r1 $tpc_r2 $reads
+            """
+        }
+    }
+}
+
+
+/*
+ * STEP 2 - Map with BWA Mem
+ */
+
+process bwamem {
+    tag "$name"
+
+    input:
+    set val(name), file(reads) from trimmed_reads
+    file(bwa_index) from bwa_index
+
+    output:
+    set val(name), file("${name}_bwa.sam") into raw_aln_sam
+    file '.command.log' into bwa_stdout
+
+
+    script:
+    rg="\'@RG\\tID:${params.run_id}\\tSM:${params.run_id}\\tPL:illumina\'"
```

Comment:
Can this not be inline in the script blocks?

Reply:
Was fixed in restructuring of input params commit.

Reply:
Hmm, I tried but the ", `, ' situation quite confused me after a while. We need the readgroups for GATK downstream though, so they need to be there.